### PR TITLE
Issue/2201

### PR DESCRIPTION
**Pull request by michael-simons**
- [ ] You have read the [Spring Data Neo4j contribution guidelines](https://github.com/spring-projects/spring-data-neo4j/blob/master/CONTRIBUTING.adoc).
- [ ] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

File: src/main/java/org/springframework/data/neo4j/core/mapping/Neo4jMappingContext.java [hunk 1]
```diff
@@ -247,13 +253,45 @@ protected Neo4jPersistentProperty createPersistentProperty(Property property, Ne
...
 		}
 		return super.addPersistentEntity(typeInformation);
 	}
 
+	/**
+	 * @param typeInformation The type to retrieve an persistent entity for
+	 * @return An optional persistent entity
+	 * @see #doGetPersistentEntity(Class)
+	 */
+	private @Nullable Neo4jPersistentEntity<?> doGetPersistentEntity(TypeInformation<?> typeInformation) {
```
Comment (meistermeier) [inline src/main/java/org/springframework/data/neo4j/core/mapping/Neo4jMappingContext.java:268]: For this method and the next: Can we have the `@Nullable` above the declaration, please.

File: src/main/java/org/springframework/data/neo4j/core/mapping/DefaultNeo4jPersistentEntity.java [hunk 2]
```diff
@@ -300,13 +305,14 @@ private void verifyDynamicLabels() {
 	 * 3. If only {@link Node#labels()} property is set, use the first one as the primary label 4. If the
 	 * {@link Node#primaryLabel()} property is set, use this as the primary label
 	 *
+	 * @param type the type of the underlying class
 	 * @return computed primary label
 	 */
-	private String computePrimaryLabel() {
+	static @Nullable String computePrimaryLabel(Class<?> type) {
```
Comment (meistermeier) [inline src/main/java/org/springframework/data/neo4j/core/mapping/DefaultNeo4jPersistentEntity.java:311]: `@Nullable` please one line above.

**Review by meistermeier — CHANGES_REQUESTED** [review]

File: src/test/java/org/springframework/data/neo4j/core/mapping/Neo4jMappingContextTest.java [hunk 3]
```diff
@@ -386,17 +491,26 @@ public Void generateId(String primaryLabel, Object entity) {
 	@Node("User")
 	static class UserNode {
 
-		@org.springframework.data.annotation.Id @GeneratedValue private long id;
+		@org.springframework.data.annotation.Id @GeneratedValue @SuppressWarnings("unused")
+		private long id;
+
+		@Relationship(type = "OWNS") @SuppressWarnings("unused")
```
Comment (meistermeier) [inline src/test/java/org/springframework/data/neo4j/core/mapping/Neo4jMappingContextTest.java:497]: Why do we start now with using `SuppressWarnings`?

**Comment by michael-simons** [comment]
Closed with 43a3047

**Pull request closed without merging**